DuckDuckGo/OnboardingExperiment/MetricBuilder/MetricBuilder.swift

```diff
@@ -0,0 +1,69 @@
+//
+//  MetricBuilder.swift
+//  DuckDuckGo
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import SwiftUI
+import class UIKit.UIScreen
+
+final class MetricBuilder<T> {
```

Comment:
The idea behind this class is to build device-specific values for the UI. For example, a scenario where the top padding for the iPhone is 100 and the iPad is 200.

This class could be extended with additional helpers such as:

`func value(iPhonePortrait: T, iPhoneLandscape: T) -> T` and so on, but I hope the design won't change much between devices and we won’t need them.

Reply:
Yup, indeed, I was thinking that I could write some tests and move this to Core or a local package so we could reuse it across the App.